In [6]:
import pandas as pd
import numpy as np

In [4]:
!pip install openpyxl
import openpyxl

In [9]:
df=pd.read_csv(r"C:\Users\kyash\Downloads\titanic.cleaned.xls")

In [15]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [19]:
a=df.shape
print("Number of rows and columns are : ",a)

Number of rows and columns are :  (891, 12)


In [20]:
Columns=df.columns
print("The name of columns are : ",Columns)

The name of columns are :  Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


In [21]:
df.info

<bound method DataFrame.info of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                   

In [24]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [25]:
(df.isnull().sum()/len(df))*100

PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64

In [27]:
df.duplicated().sum()

0

In [31]:
df["Age"].mean()

29.69911764705882

In [33]:
df["Age"].median()

28.0

In [11]:
df["Age"]=df["Age"].fillna(df["Age"].median())

In [35]:
df["Age"].isnull().sum()

0

In [12]:
df["Cabin"]=df["Cabin"].fillna("Unknown")

In [37]:
df["Cabin"].isnull().sum()

0

In [13]:
df=df.dropna(subset=["Embarked"])

In [41]:
df["Embarked"].isnull().sum()

0

In [42]:
after_clean=df.shape
print("The number of rows and columns after cleaning the dataset are : " ,after_clean)

The number of rows and columns after cleaning the dataset are :  (889, 12)


In [43]:
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Cabin          0
Embarked       0
dtype: int64

In [45]:
df.duplicated().sum()

0

In [67]:
total_passengers=len(df)
print("Number of total passengers are :",total_passengers)

Number of total passengers are : 889


In [63]:
survived=df["Survived"].value_counts()[1]
print(survived)

340


In [64]:
Not_survived=df["Survived"].value_counts()[0]
print(Not_survived)

549


In [66]:
survival_rate=(survived/total_passengers)*100
print("survival rate of passengers are : ",survival_rate)

survival rate of passengers are :  38.24521934758155


In [71]:
df.groupby("Sex")["Survived"].sum()

Sex
female    231
male      109
Name: Survived, dtype: int64

In [73]:
df.groupby("Sex")["Survived"].mean()*100

Sex
female    74.038462
male      18.890815
Name: Survived, dtype: float64

In [74]:
df["Pclass"].value_counts().sort_index()

Pclass
1    214
2    184
3    491
Name: count, dtype: int64

In [75]:
df.groupby("Pclass")["Survived"].sum()

Pclass
1    134
2     87
3    119
Name: Survived, dtype: int64

In [76]:
df.groupby("Pclass")["Survived"].mean()*100

Pclass
1    62.616822
2    47.282609
3    24.236253
Name: Survived, dtype: float64

In [78]:
df.groupby(["Pclass","Sex"]).size()
    

Pclass  Sex   
1       female     92
        male      122
2       female     76
        male      108
3       female    144
        male      347
dtype: int64

In [80]:
df.groupby(["Pclass","Sex"])["Survived"].sum()
    

Pclass  Sex   
1       female    89
        male      45
2       female    70
        male      17
3       female    72
        male      47
Name: Survived, dtype: int64

In [81]:
df.groupby(["Pclass", "Sex"])["Survived"].mean() * 100

Pclass  Sex   
1       female    96.739130
        male      36.885246
2       female    92.105263
        male      15.740741
3       female    50.000000
        male      13.544669
Name: Survived, dtype: float64

In [5]:
df["Age"].mean()

29.69911764705882

In [6]:
df["Age"].median()

28.0

In [7]:
df["Age"].min()

0.42

In [8]:
df["Age"].max()

80.0

In [10]:
bins=[0,12,19,59,80]

labels=["Child","Teenager","Adult","Senior"]

df["AgeGroup"] = pd.cut(df["Age"], bins=bins, labels=labels, include_lowest=True)

In [12]:
df["AgeGroup"].value_counts().sort_index()

AgeGroup
Child        69
Teenager     95
Adult       524
Senior       26
Name: count, dtype: int64

In [15]:
df.groupby("AgeGroup",observed=True)["Survived"].mean()*100

AgeGroup
Child       57.971014
Teenager    41.052632
Adult       38.931298
Senior      26.923077
Name: Survived, dtype: float64

In [16]:
df["Fare"].mean()

32.204207968574636

In [17]:
df["Fare"].median()

14.4542

In [18]:
df["FareGroup"] = pd.qcut(
    df["Fare"],
    q=3,
    labels=["Low Fare", "Medium Fare", "High Fare"]
)

In [19]:
df["FareGroup"].value_counts().sort_index()

FareGroup
Low Fare       308
Medium Fare    288
High Fare      295
Name: count, dtype: int64

In [20]:
df.shape

(891, 14)

In [21]:
df["Age"] = df["Age"].fillna(df["Age"].median())

In [22]:
df["Cabin"]=df["Cabin"].fillna("Unknown")

In [23]:
df.shape

(891, 14)

In [24]:
df.dropna(subset=["Embarked"])

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,AgeGroup,FareGroup
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,Unknown,S,Adult,Low Fare
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C,Adult,High Fare
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,Unknown,S,Adult,Low Fare
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S,Adult,High Fare
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,Unknown,S,Adult,Low Fare
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,Unknown,S,Adult,Medium Fare
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,B42,S,Teenager,High Fare
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,28.0,1,2,W./C. 6607,23.4500,Unknown,S,NaN,Medium Fare
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C,Adult,High Fare


In [30]:
df.groupby("FareGroup",observed=True)["Survived"].mean()*100

FareGroup
Low Fare       19.805195
Medium Fare    40.277778
High Fare      55.932203
Name: Survived, dtype: float64

In [27]:
df["Embarked"].value_counts()

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [33]:
df.groupby("Embarked",observed=True)["Survived"].mean()*100

Embarked
C    55.357143
Q    38.961039
S    33.695652
Name: Survived, dtype: float64

Insight 1: Female passengers had a significantly higher survival rate than male passengers, with 74.03% of females surviving compared with 18.91% of males.

Insight 2: Survival rate increased consistently with ticket fare. High-fare passengers had the highest survival rate at 55.93%, compared with 40.28% for medium-fare and 19.81% for low-fare passengers.

Insight 3: Survival rates generally decreased with increasing age. Children had the highest survival rate at 57.97%, while seniors had the lowest at 26.92%.

Insight 4: The overall survival rate was 38.25%, with 340 out of 889 passengers surviving. This indicates that fewer than half of the analyzed passengers survived.

Insight 5: Passenger class and gender were strongly associated with survival. First-class female passengers had the highest survival rate at 96.74%, while third-class male passengers had the lowest at 13.54%.

In [36]:
df.to_csv("titanic.cleaned.csv",index=False)

In [14]:
df.shape

(889, 14)

In [15]:
df.to_excel("titanic_cleaned.xlsx", index=False)

In [16]:

import os
os.getcwd()

'C:\\Users\\kyash'